src/inference/include/openvino/runtime/core.hpp

```diff
@@ -65,27 +65,6 @@ class OPENVINO_RUNTIME_API Core {
      */
     std::map<std::string, Version> get_versions(const std::string& device_name) const;
 
-#ifdef OPENVINO_ENABLE_UNICODE_PATH_SUPPORT
-    /**
-     * @brief Reads models from IR / ONNX / PDPD / TF / TFLite file formats.
-     * @param model_path Path to a model.
-     * @param bin_path Path to a data file.
-     * For IR format (*.bin):
-     *  * if `bin_path` is empty, will try to read a bin file with the same name as xml and
-     *  * if the bin file with the same name is not found, will load IR without weights.
-     * For the following file formats the `bin_path` parameter is not used:
-     *  * ONNX format (*.onnx)
-     *  * PDPD (*.pdmodel)
-     *  * TF (*.pb, *.meta, SavedModel directory)
-     *  * TFLite (*.tflite)
-     * @param properties Optional map of pairs: (property name, property value) relevant only for this read operation.
-     * @return A model.
-     */
-    std::shared_ptr<ov::Model> read_model(const std::wstring& model_path,
```

Comment:
The general direction should be to remove string versions (and is good) but they cannot be removed without
deprecation first  (maybe only std::string versions could be removed).

The implementation should be done like std::filesystem::Path keeps main implementation and std::(w)string are just wrappers to create paths objects. The version with wstring can be useful to apply our internal util for wstring conversion to create paths objects. But it will requires testing which option is the best.

The internal classes/functions can use path object without deprecation, just requires proper handling